viewing_party/party.py

```diff
@@ -1,23 +1,131 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {
+        "title": title, 
+        "genre": genre,
+        "rating": rating
+    }
+
+    if None in movie.values():
+        return None
+
+    else:
+        return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movie in list(user_data["watchlist"]):
+        if movie["title"] == movie_title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    if not user_data["watched"]:
+        return 0
 
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+
+    average = sum / len(user_data["watched"])
+    return average

```

Comment:
Nice job calculating the average yourself by accumulating the total and dividing by the number of movies. It's good practice to think how to do these kinds of operations ourselves. The next time you need to find an average, try looking into other built-in python functions that might be useful!